manage

```diff
@@ -696,11 +696,15 @@ test.pyright() {
     node.env.devtools
     # We run Pyright in the virtual environment because Pyright
     # executes "python" to determine the Python version.
-    pyenv.cmd npx --no-install pyright -p pyrightconfig-ci.json
+    build_msg TEST "[pyright] suppress warnings related to intentional monkey patching"
+    pyenv.cmd npx --no-install pyright -p pyrightconfig-ci.json \
```

Comment:
When the pyenv isn't installed yet the "suppress warnings" message is printed long before the warnings are actually suppressed.

Reply:
Yes, what do you suggest?

Reply:
How about:

```
pyenv.activate
build_msg TEST "[pyright] suppress warnings related to intentional monkey patching"
npx --no-install pyright ...
```

Reply:
> How about:
>    pyenv.activate  

and what about the build messages above, its also printed "long before" ..

> it to be documented in the code.

I think this requirement is fulfilled.

Reply:
>what about the build messages above

It makes sense to me ... in a log it makes sense to print why sth is installed before you install it.

Reply:
If I understand correctly, the issue appears when someone starts `make test.pyright` without a virtualenv or if the deppendencies are changed? Otherwise there is no issue, for example `make test` is going to first install the dependencies?

Reply:
Its not really an issue .. its common that (underlying) tools spit out their messages and this might cause that there is a distance between one "build" message and the next "build" message.

Here in this case @not-my-profile suggested ...

```
pyenv.activate
build_msg TEST "[pyright] suppress warnings related to intentional monkey patching"
npx --no-install pyright ...
```

instead of ..

```
build_msg TEST "[pyright] suppress warnings related to intentional monkey patching"
pyenv.cmd npx --no-install pyright ...
```

Consequently, this would mean we can replace nearly all `pyenv.cmd` what makes no sense to me .. tbh I do not understand this discussion and merged this PR.

@not-my-profile if you think there is a change needed, please open a PR for.

Reply:
Right ... that could have been avoided using a subshell. But you're right that it's not really that important.